code/workspaces/infrastructure-api/src/controllers/stackController.js

```diff
@@ -205,6 +206,13 @@ const createStackValidator = [
       }
       return true;
     }),
+  check('version', 'valid version must be specified')
+    .custom((value, { req }) => {
+      if (Object.prototype.hasOwnProperty.call(req.body, 'version')) {
+        return versionList(req.body.type).includes(req.body.version);
+      }
+      return true;
+    }),
```

Comment:
Could this be simplified with `optional`? https://express-validator.github.io/docs/validation-chain-api.html#optionaloptions

_**note:** I've not tried this out!_

```js
check('version', 'valid version must be specified')
  .optional()
  .custom((value, { req }) => versionList(req.body.type).includes(req.body.version)),
```

Might also be nice if it is possible to provide a list of the valid versions too. This might be possible with something like 

```js
check('version')
  .optional()
  .custom((value, { req }) => versionList(req.body.type).includes(req.body.version))
  .withMessage((value, { req }) => `Must be one of ${versionList(req.body.type)}.`),
```